src/main/java/at/hannibal2/skyhanni/utils/TimeUtils.kt

```diff
@@ -154,6 +156,25 @@ object TimeUtils {
         ).orEmpty()
     }
 
+    fun SimpleTimeMark.formatForHoppity(): Pair<String, Boolean> =
+        if (SkyHanniMod.feature.event.hoppityEggs.eventSummary.liveDisplay.dateTimeFormat == RELATIVE)
+            Pair(passedSince().absoluteValue.format(maxUnits = 2), false)
+        else {
+            val timeNow = SimpleTimeMark.now().toLocalDateTime()
+            val timeThen = toLocalDateTime()
+
+            val yearDiff = timeThen.year - timeNow.year
+            val monthDiff = timeThen.monthValue - timeNow.monthValue
+            val dayDiff = timeThen.dayOfMonth - timeNow.dayOfMonth
+
+            val dateFormat = when {
+                yearDiff == 0 && monthDiff == 0 && dayDiff == 0 -> "HH:mm:ss"
+                (yearDiff == 0 && monthDiff == 0) || (yearDiff == 0) -> "MM-dd HH:mm"
+                else -> "yyyy-MM-dd HH:mm"
+            }
+            Pair(formattedDate(dateFormat), true)
+        }
+
```

Comment:
I dont like the idea of this being a public extension function that can be used everywhere. Also should be in some sort of hoppity class instead of time utils

Reply:
https://github.com/hannibal002/SkyHanni/pull/2973#discussion_r1875015418

Reply:
:( Hannibal

Ideally the extension function goes back to where you had it, but it is split so that if it is relative time then it returns that and otherwise it calls a new time utils function called format absolute time or whatever.
@hannibal002 what is your thoughts on this as i do not feel there should be a hoppity specific function in time utils or one that extends simple time mark. Maybe when you requested the original changes you did not realise there was hoppity specific code?